pepper-apis/dsm-core/src/test/java/org/broadinstitute/dsm/service/adminoperation/LegacyKitResampleServiceTest.java

```diff
@@ -0,0 +1,222 @@
+import org.broadinstitute.dsm.db.DDPInstance;
+import org.broadinstitute.dsm.db.KitRequestShipping;
+import org.broadinstitute.dsm.db.dao.ddp.instance.DDPInstanceDao;
+import org.broadinstitute.dsm.db.dao.ddp.kitrequest.KitRequestDao;
+import org.broadinstitute.dsm.db.dto.ddp.instance.DDPInstanceDto;
+import org.broadinstitute.dsm.db.dto.ddp.participant.ParticipantDto;
+import org.broadinstitute.dsm.juniperkits.TestKitUtil;
+import org.broadinstitute.dsm.util.ElasticSearchUtil;
+import org.broadinstitute.dsm.util.ElasticTestUtil;
+import org.broadinstitute.dsm.util.MedicalRecordTestUtil;
+import org.broadinstitute.dsm.util.TestParticipantUtil;
+import org.junit.After;
+import org.junit.AfterClass;
+import org.junit.Assert;
+import org.junit.BeforeClass;
+import org.junit.Test;
+
+@Slf4j
+public class LegacyKitResampleServiceTest  extends DbAndElasticBaseTest {
+    private static final String instanceName = "kit_resample_service";
+    private static final String shortId = "PT_SHORT";
+    private static final String legacyShortId = "LEGACY_SHORT";
+    private static String esIndex;
+    private static DDPInstanceDto ddpInstanceDto;
+    private static DDPInstanceDao ddpInstanceDao = new DDPInstanceDao();
+    private static String ddpParticipantId = "PT_SAMPLE_QUEUE_TEST";
+    private static ParticipantDto participantDto = null;
+    private static int participantCounter = 0;
+    private static Pair<ParticipantDto, String> legacyParticipantPair;
+    public static TestKitUtil testKitUtil;
+    private static final String newCollaboratorSampleId = "NEW_SAMPLE_ID";
+    private static final String newCollaboratorParticipantId = "NEW_PARTICIPANT_ID";
+
+    private static final String oldCollaboratorSampleId = "OLD_SAMPLE_ID";
+    private static final String oldCollaboratorParticipantId = "OLD_PARTICIPANT_ID";
+    private static final String ddpKitRequestId = "RESAMPLE_KIT_REQUEST_ID";
+    private static List<ParticipantDto> participants = new ArrayList<>();
+    private static List<String> createdKits = new ArrayList<>();
+
+    private static LegacyKitResampleService legacyKitResampleService = new LegacyKitResampleService();
+    private static DDPInstance ddpInstance;
+
+    @BeforeClass
+    public static void doFirst() {
+        testKitUtil = new TestKitUtil(instanceName, instanceName, "resample", instanceName, "SALIVA", null);
+        testKitUtil.setupInstanceAndSettings();
+        esIndex = ElasticTestUtil.createIndex(instanceName, "elastic/lmsMappings.json", null);
+        ddpInstanceDto = ddpInstanceDao.getDDPInstanceByInstanceName(instanceName).orElseThrow();
+        ddpInstanceDto.setEsParticipantIndex(esIndex);
+        ddpInstanceDao.updateEsParticipantIndex(ddpInstanceDto.getDdpInstanceId(), esIndex);
+        ddpParticipantId = TestParticipantUtil.genDDPParticipantId(ddpParticipantId);
+        legacyParticipantPair = TestParticipantUtil.createLegacyParticipant(ddpParticipantId, participantCounter++, ddpInstanceDto,
```

Comment:
Typically you would want to create participants in the tests and clean them up in the after method, but if you are only using then in one test this should work.